alexandria/core/validations.py

```diff
@@ -1,13 +1,48 @@
+import logging
 from mimetypes import guess_type
 
 import magic
+from clamdpy import ClamdNetworkSocket
+from django.conf import settings
 from django.utils.translation import gettext_lazy as _
-from django_clamd.validators import validate_file_infection
 from generic_permissions.validation import validator_for
 from rest_framework.exceptions import ValidationError
 
 from alexandria.core.models import File
 
+log = logging.getLogger(__name__)
+
+
+def validate_file_infection(file):
+    # inspired by django-clamd
+    if file is None or not settings.CLAMD_ENABLED:
+        return
+
+    scanner = ClamdNetworkSocket(settings.CLAMD_TCP_ADDR, settings.CLAMD_TCP_SOCKET)
+    try:
+        file.seek(0)
+        result = scanner.instream(file)
+    except IOError:
+        # Ping the server if it fails than the server is down
+        scanner.ping()
+
+        # Server is up. This means that the file is too big.
+        log.warning(
+            "The file is too large for ClamD to scan it. Bytes Read {}".format(
+                file.tell()
+            )
+        )
+        raise ValidationError(
+            _("Malware scan could not completed. Please try again later."),
```

Comment:
Needs a `manage.py makemessages` run I think. Also, potentially reword a bit..

```suggestion
            _("Malware scan could not completed. Please notify the responsible administrator."),
```

(Rewording, as just trying again will not help in this case, and the ClamD config likely needs to change)

Reply:
Applied

Reply:
Looked at clamdpy API a bit more and it handles too large files and other errors itself already.
Therefore the try catch was removed

Reply:
Even better :-)